framework/include/utils/MooseTypes.h

```diff
@@ -680,8 +680,9 @@ enum EigenSolveType
   EST_JACOBI_DAVIDSON, ///< Jacobi-Davidson
   EST_NONLINEAR_POWER, ///< Nonlinear inverse power
   EST_NEWTON, ///< Newton-based eigensolver with an assembled Jacobian matrix (fully coupled by default)
-  EST_PJFNK, ///< Preconditioned Jacobian-free Newton Krylov
-  EST_JFNK   ///< Jacobian-free Newton Krylov
+  EST_PJFNK,   ///< Preconditioned Jacobian-free Newton Krylov
+  EST_PJFNKMO, ///< The same as PJFNK except that matrix-vector multiplication is employed to replace residual evaluation in linear solver
```

Comment:
MO: Matrix-Only

Reply:
So Jacobian-Free Matrix-Only ... I have to say that seems like a natural contradiction (although I understand that you are using matrix P and no matrix J). Is Jacobian-Free Matrix-Only terminology that is present in literature?

Reply:
It is good that we've used the Jacobian-free terminology from the start, as opposed to matrix-free, otherwise this new solve type would be Matrix-Free Matrix-Only 😆

Reply:
> Is Jacobian-Free Matrix-Only terminology that is present in literature?
 
 No. We just come up this crazy name even though I do not really like it but I do not have a better one 
 
 Maybe I should call  it FastPJFNK

Reply:
Haha no, I think the current name conveys more accurate meaning than `FastPJFNK` (even if you have to think about it)